images/ad-server/Containerfile

```diff
@@ -1,35 +1,52 @@
-FROM quay.io/samba.org/sambacc:latest AS builder
-ARG SAMBACC_VER=master
-ARG SAMBACC_REPO=https://github.com/samba-in-kubernetes/sambacc
-
-# The SAMBACC_VER value controls what version of sambacc we'll be building.
-# This can be a branch name, a tag, or a version hash.  When building tagged
-# versions of samba-containers we should use a sambacc tag.
-# SAMBACC_DISTNAME controls where the script will output a built wheel.
-RUN SAMBACC_DISTNAME=latest \
-    /usr/local/bin/build.sh ${SAMBACC_VER} ${SAMBACC_REPO}
-
-FROM registry.fedoraproject.org/fedora:36
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
```

Comment:
I disagree with this approach. Different base os-image implies different installer-program (dnf, zypper, apt..) as well as different list of packages names.  It requires adding `if`s within the `Containerfile` which makes it hard-to-read and hard-to-maintain over time. 

IMHO, a better approach would be to have multiple Container files per base OS and pass it as argument to top-level `Makefile`.